src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -51,39 +51,56 @@ def total_avg_trade_price(markets):
     )
 
 
-def energy_bills(area, past_market_types, from_slot=None, to_slot=None):
+def _store_bought_trade(result_dict, trade_offer):
+    result_dict['bought'] += trade_offer.energy
+    result_dict['spent'] += trade_offer.price
+    result_dict['total_energy'] += trade_offer.energy
+    result_dict['total_cost'] += trade_offer.price
+
+
+def _store_sold_trade(result_dict, trade_offer):
+    result_dict['sold'] += trade_offer.energy
+    result_dict['earned'] += trade_offer.price
+    result_dict['total_energy'] -= trade_offer.energy
+    result_dict['total_cost'] -= trade_offer.price
+
+
+def energy_bills(area, past_market_types, external_trades, from_slot=None, to_slot=None):
```

Comment:
Not part of this story, bit as far as I see, the from_slot to_slot functionality is not used at the moment. Is this already implemented for the future?

Reply:
It used to be useful, in order to calculate the energy bills for a time range, through the REST API of d3a. Now that the REST API has been removed, I think we can remove this, let me know if you agree and I will get rid of it. Thanks

Reply:
I agree, thanks!

Reply:
Thanks, removed this